abi-bindings/go/packer/packer_test.go

```diff
@@ -0,0 +1,204 @@
+	"ProtocolRegistryEntry":  &teleporterregistry.ProtocolRegistryEntry{},
+}
+
+func findAllImplementers(t *testing.T) []string {
+	cfg := &packages.Config{
+		Mode: packages.NeedName | packages.NeedFiles | packages.NeedSyntax | packages.NeedTypes | packages.NeedTypesInfo,
+		Fset: fs,
+	}
+	interfacePkg, err := packages.Load(cfg, currentPackagePath)
+	require.NoError(t, err)
+
+	var iface *types.Interface
+	// find the interface in the currnet package
+	for _, pkg := range interfacePkg {
+		for _, obj := range pkg.Types.Scope().Names() {
+			if obj == interfaceName {
+				ifaceObj := pkg.Types.Scope().Lookup(obj)
+				if ifaceObj == nil {
+					continue
+				}
+				if typ, ok := ifaceObj.Type().Underlying().(*types.Interface); ok {
+					iface = typ
+					break
+				}
+			}
+		}
+	}
+	require.NotNilf(t, iface, "Interface %s not found in package %s", interfaceName, currentPackagePath)
+
+	pkgs, err := packages.Load(cfg, packagePath)
+	require.NoError(t, err)
+
+	allImplementers := []string{}
+
+	for _, pkg := range pkgs {
+		for _, file := range pkg.Syntax {
+			// Inspect types in the package
+			for _, decl := range file.Decls {
+				genDecl, ok := decl.(*ast.GenDecl)
+				if !ok {
+					continue
+				}
+				for _, spec := range genDecl.Specs {
+					// We have a Type spec
+					typeSpec, ok := spec.(*ast.TypeSpec)
+					if !ok {
+						continue
+					}
+					// Find object matching the TypeSpec name in the package scope
+					obj := pkg.Types.Scope().Lookup(typeSpec.Name.Name)
+					if obj == nil {
+						continue
+					}
+					// Check if the typed object we have is a struct
+					_, ok = obj.Type().Underlying().(*types.Struct)
+					if !ok {
+						continue
+					}
+					// Lookup the object by name and cast it to types.Named
+					typeName := typeSpec.Name.Name
+					typeObj := pkg.Types.Scope().Lookup(typeName)
+					if typeObj == nil {
+						continue
+					}
+					named, ok := typeObj.Type().(*types.Named)
+					if !ok {
+						continue
+					}
+					// Finally check if the named struct we have implements the interface
+					if types.Implements(types.NewPointer(named), iface) {
+						allImplementers = append(allImplementers, named.Obj().Name())
+					}
+				}
+			}
+		}
+	}
+	return allImplementers
+}
+
+func TestExhaustivePacking(t *testing.T) {
+	implementers := findAllImplementers(t)
+	for _, structName := range implementers {
+		packerType, ok := packerTypes[structName]
+		require.True(
+			t,
+			ok,
+			fmt.Sprintf("Struct %s not found in packer_test.go packerTypes map. Add it to check for exhaustiveness", structName),
+		)
+
+		randomizedStruct, err := randomizeStruct(packerType)
+		require.NoError(t, err)
+
+		v1 := randomizedStruct.(ABIPacker)
+		encoded, err := v1.Pack()
+		require.NoError(t, err)
+		v2 := reflect.New(reflect.TypeOf(v1).Elem()).Interface().(ABIPacker)
+		err = v2.Unpack(encoded)
+		require.NoError(t, err)
+		require.Equal(t, v1, v2)
+	}
+}
+
+func randomizeStruct(packerStruct interface{}) (interface{}, error) {
```

Comment:
I'd recommend renaming these to `populateStruct` and `populateField` so that there's no misconceptions that the test relies on randomness.